Tests/AnyCodableTests/AnyEncodableTests.swift

```diff
@@ -66,9 +66,47 @@ class AnyEncodableTests: XCTestCase {
         XCTAssert(encodedJSONObject["integer"] is Int)
         XCTAssert(encodedJSONObject["double"] is Double)
     }
+    
+    #if swift(>=5.0)
+    func testStringInterpolationEncoding() throws {
+        let dictionary: [String: AnyEncodable] = [
+            "boolean": "\(true)",
+            "integer": "\(1)",
+            "double": "\(3.141592653589793)",
+            "string": "\("string")",
+            "array": "\([1, 2, 3])",
+        ]
+
+        let encoder = JSONEncoder()
 
+        let json = try encoder.encode(dictionary)
+        let encodedJSONObject = try JSONSerialization.jsonObject(with: json, options: []) as! NSDictionary
+
+        let expected = """
+        {
+            "boolean": "true",
+            "integer": "1",
+            "double": "3.141592653589793",
+            "string": "string",
+            "array": "[1, 2, 3]",
+        }
+        """.data(using: .utf8)!
+        let expectedJSONObject = try JSONSerialization.jsonObject(with: expected, options: []) as! NSDictionary
+
+        XCTAssertEqual(encodedJSONObject, expectedJSONObject)
+    }
+    #endif
+
+    #if swift(>=5.0)
+    static var allTests = [
+        ("testJSONEncoding", testJSONEncoding),
+        ("testEncodeNSNumber", testEncodeNSNumber),
+        ("testStringInterpolationEncoding", testStringInterpolationEncoding),
+    ]
+    #else
     static var allTests = [
         ("testJSONEncoding", testJSONEncoding),
         ("testEncodeNSNumber", testEncodeNSNumber),
     ]
+    #endif
```

Comment:
not needed if "discovered" runtime `swift test --enable-test-discovery`
or generated `swift test --generate-linuxmain`

Reply:
Correct. The only reason I'm keeping around `allTests` is to keep Swift 4 compatibility (which is probably not a big concern anymore, but it's hard to justify dropping support just for `--enable-test-discovery`).